src/proto/tagpost.proto

```diff
@@ -7,17 +7,68 @@ option java_package = "com.google.tagpost";
 option java_outer_classname = "TagPostProto";
 
 service TagpostService {
+  // dummy rpc service
   rpc FetchMessage(FetchMessageRequest) returns (FetchMessageResponse) {}
+  // fetch all threads by primary tag
+  rpc FetchThreadsByTag(FetchThreadsByTagRequest) returns (FetchThreadsByTagResponse) {}
+  // fetch all comments under one thread
+  rpc FetchCommentsUnderThread(FetchCommentsUnderThreadRequest) returns (FetchCommentsUnderThreadResponse) {}
+  // fetch all comments by a set of tags
+  rpc FetchCommentsByTags(FetchCommentsByTagsRequest) returns (FetchCommentsByTagsResponse) {}
+}
+
+message Tag {
+  string tag_name = 1;
+}
+
+message TagpostThread {
+  int64 thread_id = 1;
+  string primary_tag = 2;
+}
+
+message Comment {
+  int64 comment_id = 1;
+  int64 thread_id = 2;
+  string comment_content = 3;
+  string timestamp = 4;
+  Tag primary_tag = 5;
+  string username = 6;
+  repeated Tag extra_tags = 7;
 }
 
 message FetchMessageRequest {
   string notice_key = 1;
   string request_id = 2;
   string user = 3;
-  string project = 4; 
+  string project = 4;
 }
 
 message FetchMessageResponse {
   string message = 1;
   repeated string comments = 2;
 }
+
+message FetchThreadsByTagRequest{
+  Tag tag = 1;
+}
+
+message FetchThreadsByTagResponse{
+  TagpostThread thread = 1;
+}
+
+message FetchCommentsUnderThreadRequest{
+  TagpostThread thread = 1;
```

Comment:
A thread is uniquely identified by its `thread_id`. However by embedding the whole `TagpostThread` message in the request message, `primary_tag` is also a part of the request.

Please either document how `primary_tag` affects the response, or use only `thread_id` in the request message.

Reply:
Fixed, now changed to use thread_id in the request